### ISSUE TITLE
reset password email not received by end user: case sensitive

### ISSUE THREAD

**Issue by sbarbosadataverse** [IQSS/dataverse#2879]
https://help.hmdc.harvard.edu/Ticket/Display.html?id=231097 

RT ticket attached for FU.
User tried several times, no reset email received, he included a screenshot indicating the email being used, and the message indicating reset link was on the way

**Comment by pdurbin** [comment]
In production, it's possible to grep through the Glassfish `server.log` to get more diagnostics on what went wrong but as of this writing we have three Glassfish servers. I think we should consider logging password resets to the database and exposing this information to a superuser GUI so that people like @sbarbosadataverse aren't flying blind trying to help people.

**Comment by pdurbin** [comment]
Another suggestion would be to have the logs for all three Glassfish servers get fed into something like Splunk or similar (i.e. [Logstash](https://www.elastic.co/products/logstash)) so we could search across all the logs at once. Probably @sbarbosadataverse wouldn't do this herself though. Maybe. It depends on how arcane the interface is and various security considerations.

Another suggestion (but I like the other two more) is to use something a tool that lets you execute the same command (i.e. grep) across all three Glassfish servers at once. The one I've used in the past is `gsh` but I don't think it's especially popular these days. I'd suggest looking at https://serverfault.com/questions/2533/linux-running-the-same-command-on-many-machines-at-once and other up-to-date lists.

**Comment by pdurbin** [comment]
Reviewing the case at https://help.hmdc.harvard.edu/Ticket/Display.html?id=231097 has made me realize that the password reset feature I developed in #416 is case sensitive. D'oh!

I just tested this on the develop branch (commit d65b905) by signing up for an account with the email address "testcase@Mailinator.COM" to simulate the situation in the ticket above. Then I tried resetting the password with the original case and a lower case version. In server.log I saw the following:

- original (works): "attempted to send mail to testcase@Mailinator.COM"
- lower case (bug, no email sent): "Couldn't find single account using testcase@mailinator.com"

I'll update the title of this issue to reflect the case sensitivity bug. Obviously, we don't want users to have to care about the case they used for their email address back when they signed up.

What lead me to find the difference in case of the email address was seeing it in the action log ("info" column) vs. the builtinuser table. The following queries were helpful:

- `select * from actionlogrecord where actiontype = 'BuiltinUser';`
- `select email from builtinuser where id = 42;`
